javers-core/src/main/java/org/javers/core/metamodel/scanner/FieldBasedPropertyScanner.java

```diff
@@ -20,14 +21,22 @@ public FieldBasedPropertyScanner(AnnotationNamesProvider annotationNamesProvider
 
     @Override
     public PropertyScan scan(Class<?> managedClass) {
+        final IgnoreDeclaredProperties ignoreDeclaredPropertiesAnnotation = managedClass.getAnnotation(IgnoreDeclaredProperties.class);
         List<JaversField> fields = ReflectionUtil.getAllPersistentFields(managedClass);
         List<Property> propertyList = new ArrayList<>(fields.size());
 
-        for (JaversField field : fields) {
-            boolean hasTransientAnn = field.hasAnyAnnotation(annotationNamesProvider.getTransientAliases());
-            boolean hasShallowReferenceAnn = field.hasAnyAnnotation(annotationNamesProvider.getShallowReferenceAliases());
-            propertyList.add(new Property(field, hasTransientAnn, hasShallowReferenceAnn));
+        if (ignoreDeclaredPropertiesAnnotation != null) {
```

Comment:
this is wrong, it means that when `@IgnoreDeclaredProperties` is present -- all properties will be ignored